coalib/settings/ConfigurationGathering.py

```diff
@@ -192,9 +193,12 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
         sections = merge_section_dicts(sections, cli_sections)
 
-    for section in sections:
-        if section != 'default':
-            sections[section].defaults = sections['default']
+    for name, section in list(sections.items()):
+        if name != 'default':
+            section.set_default_section(sections)
```

Comment:
I'd move this outside the if even and just do `if name == 'default': logging.warning(...)`